components/org.wso2.carbon.extension.identity.sso.jwt/src/main/java/org/wso2/carbon/identity/sso/jwt/processor/JWTInboundRequestProcessor.java

```diff
@@ -0,0 +1,669 @@
+                ERROR_URL_PARAM_NAME);
+        if (log.isDebugEnabled()) {
+            log.debug(
+                    "Setting the query parameter names.\nJWT query parameter: " + neutralize(jwtParamName) +
+                            "\nRedirect URL query parameter: " + neutralize(redirectUrlParamName) +
+                            "\nError URL query parameter: " + neutralize(errorUrlParamName));
+        }
+        respBuilder.setJwtParamName(jwtParamName);
+        respBuilder.setRedirectUrlParamName(redirectUrlParamName);
+        respBuilder.setErrorUrlParamName(errorUrlParamName);
+
+        return respBuilder;
+    }
+
+    /**
+     * Handles the response coming after logged out.
+     *
+     * @param identityRequest The identity request
+     * @param respBuilder     The response builder
+     * @return The response builder after setting the required options
+     */
+    private JWTInboundResponse.JWTInboundResponseBuilder handleLogoutResult(
+            IdentityRequest identityRequest, JWTInboundResponse.JWTInboundResponseBuilder respBuilder) {
+
+        String logoutUrl = getPropertyValue(identityRequest, JWTInboundConstants.SPBasedConfigs.LOGOUT_URL);
+        if (log.isDebugEnabled()) {
+            log.debug(
+                    "Non-authenticated session. Treating as the request coming after logout.\n" +
+                            "Setting the logout URL: " + neutralize(logoutUrl));
+        }
+        // Validate and set Logout URL
+        if (StringUtils.isNotBlank(logoutUrl)) {
+            respBuilder.setToken(null);
+            respBuilder.setLogoutUrl(logoutUrl);
+        } else {
+            String msg = "Mandatory configuration: Logout URL is not configured.";
+            log.error(msg, new JWTIdentityException(msg));
+            return JWTInboundUtil.sendToRetryPage(JWTInboundConstants.ErrorMessages.MISCONFIGURATION_STATUS,
+                    JWTInboundConstants.ErrorMessages.MISCONFIGURATION_MESSAGE);
+        }
+        return respBuilder;
+    }
+
+    private String getPropertyValue(IdentityRequest request, String property) {
+
+        String propertyValue = null;
+        Map<String, Property> props = getInboundAuthenticatorPropertyArray(request);
+        for (Object obj : props.entrySet()) {
+            Map.Entry pair = (Map.Entry) obj;
+            if (property.equals(pair.getKey())) {
+                Property prop = (Property) pair.getValue();
+                propertyValue = prop.getValue();
+            }
+        }
+        return propertyValue;
+    }
+
+    /**
+     * This method is used for retrieving the values for the properties from JWTInboundAuthConfig which will
+     * have been set in the Identity Server's Service Provider settings.
+     *
+     * @param request the identity request object
+     * @return a map of available properties and their values
+     */
+    private Map<String, Property> getInboundAuthenticatorPropertyArray(IdentityRequest request) {
```

Comment:
Looks like this only needs the tenantDomain from the request. Lets just pass the tenant domain to the method as a string. It is a bit confusing when looking at places where the method is used as it seems like we are extracting values from the request itself.

Reply:
Fixed